@shared/api/helpers/extensionMessaging.ts

```diff
@@ -67,11 +68,21 @@ export const sendMessageToContentScript = (msg: Msg): Promise<Response> => {
 
 export const sendMessageToBackground = async (msg: Msg): Promise<Response> => {
   let res;
+
+  // grab the active public key from the popup and pass it along with every message.
+  // this will be used to determine if the popup is open to the correct account.
+  const extensionState = popupStoreFacade.currentStore?.getState();
```

Comment:
I was looking for a solution where we didn't have to remember to always send the "active" public key to the background with every request as that would be easy to forget in the future.

This solution automatically appends the active public key to every request that goes to the message listener.